src/rust_helpers/number.rs

```diff
@@ -0,0 +1,40 @@
+use std::hash;
+
+#[derive(Clone, Copy, Debug)]
+pub struct Number(f64);
```

Comment:
It doesn't really matter at this point, but is this accurate to how JS stores numbers? (I don't know if under the hood it switches between integer and floating point representations?)

Reply:
Right I'm really not sure. From what I've seen so far the important part will be that it can equality-compare different number literal (types) which seems like `f64` should likely work for